### PR TITLE
[#ICC-233] Add permission check for template value on Notify

### PR DESCRIPTION
**Pull request by fabriziopapi**
The Notify enpoint will profile the access to a specific notification template based on the user group. With this implementation, we could restrict the access to a specific notification template only to a subset of Services.
In particular, the Reminder service will be profiled to send REMINDER notification type only, and all other services will be profiled to send MESSAGE notification type only.
In the future, we will have more templates used only by  specific service (like Firma con IO).

#### List of Changes

add middlewares to Notify handler

#### Motivation and Context

We need to control the notify text sent to the user without letting the service input a free text: with the template profiling we can provide a wide range of possibile notification text and control which service could use them.

#### How Has This Been Tested?

unit test

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: Notify/handler.ts [hunk 1]
```diff
@@ -210,7 +215,23 @@ export const Notify = (
...
+    ),
+    AzureAllowBodyPayloadMiddleware(
+      t.interface({
+        notification_type: t.union([
+          t.literal(NotificationTypeEnum.REMINDER_PAYMENT),
+          t.literal(NotificationTypeEnum.REMINDER_PAYMENT_LAST),
+          t.literal(NotificationTypeEnum.REMINDER_READ)
+        ])
+      }),
+      new Set([UserGroup.ApiReminderNotify])
```
Comment (gquadrati) [inline Notify/handler.ts:233]: Just an idea, we could extract two codec, `MessageNotificationInfo` and `ReminderNotificationInfo`

**Review by gquadrati — DISMISSED** [review]
LGTM

**Review by AleDore — COMMENTED** [review]
LGTM as i would extract specific types as suggested by @gquadrati :)

**Comment by AleDore** [comment]
@fabriziopapi : please note that code review fails :)

**Comment by gquadrati** [comment]
> @fabriziopapi : please note that code review fails :)

It was just an error running the test.

**Review by AleDore — APPROVED** [review]

File: Notify/handler.ts (hunk 1 above)
Comment (fabriziopapi) [reply Notify/handler.ts]: done, thank you!

**Review by AleDore — APPROVED** [review]

**Pull request merged**